examples/usdc-transfer-app/src/app/page.tsx

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// docs::https://docs.sui.io/guides/developer/stablecoins
+// docs::#setup
+'use client';
+
+import { SuiClientProvider, useSuiClient } from '@mysten/dapp-kit';
+import { bcs } from '@mysten/sui/bcs';
+import { Transaction } from '@mysten/sui/transactions';
+import { ConnectButton, useWalletKit, WalletKitProvider } from '@mysten/wallet-kit';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { useEffect, useState } from 'react';
+
+// Define the network we're connecting to (testnet in this case)
+const networks = {
+	testnet: { url: 'https://fullnode.testnet.sui.io:443' },
+};
+
+// Create a new QueryClient for managing and caching asynchronous queries
+const queryClient = new QueryClient();
+// docs::/#setup
+
+// Define the USDC token type on Sui Testnet
+// This is the unique identifier for the USDC token on Sui
+const USDC_TYPE = '0xa1ec7fc00a6f40db9693ad1415d0c193ad3906494428cf252621037bd7117e29::usdc::USDC';
+
+function HomeContent() {
+	// docs::#state
+	// Use the wallet kit to get the current account and transaction signing function
+	const { currentAccount, signAndExecuteTransactionBlock } = useWalletKit();
+	// Get the Sui client for interacting with the Sui network
+	const suiClient = useSuiClient();
+	const [connected, setConnected] = useState(false);
+	const [amount, setAmount] = useState('');
+	const [recipientAddress, setRecipientAddress] = useState('');
+	const [txStatus, setTxStatus] = useState('');
+	// docs::/#state
+
+	// docs::#useeffect
+	// Update the connection status when the current account changes
+	useEffect(() => {
+		setConnected(!!currentAccount);
+	}, [currentAccount]);
+	// docs::/#useeffect
+
+	const handleSendTokens = async () => {
+		if (!currentAccount || !amount || !recipientAddress) {
+			setTxStatus('Please connect wallet and fill in all fields');
+			return;
+		}
+		try {
+			// Fetch USDC coins owned by the current account
+			// This uses the SuiClient to get coins of the specified type owned by the current address
+			const { data: coins } = await suiClient.getCoins({
+				owner: currentAccount.address,
+				coinType: USDC_TYPE,
+			});
+			if (coins.length === 0) {
+				setTxStatus('No USDC coins found in your wallet');
+				return;
+			}
+			// Create a new transaction block
+			// Transaction is used to construct and execute transactions on Sui
+			const tx = new Transaction();
+			// Convert amount to smallest unit (6 decimals)
+			const amountInSmallestUnit = BigInt(parseFloat(amount) * 1_000_000);
+			// Convert amount to serialized BCS
+			const serializedAmount = bcs.u64().serialize(amountInSmallestUnit).toBytes();
+			// Split the coin and get a new coin with the specified amount
+			// This creates a new coin object with the desired amount to be transferred
+			const [coin] = tx.splitCoins(coins[0].coinObjectId, [tx.pure(serializedAmount)]);
```

Comment:
I think it's better to skip manual serialization since it can be done simpler.
```suggestion
			// Convert amount to serialized BCS
			const serializedAmount = bcs.u64().serialize(amountInSmallestUnit).toBytes();
			// Split the coin and get a new coin with the specified amount
			// This creates a new coin object with the desired amount to be transferred
			const [coin] = tx.splitCoins(coins[0].coinObjectId, [tx.pure.u64(amountInSmallestUnit)]);
```